mlir/utils/performance/analysis/quickTuningGen.py

```diff
@@ -0,0 +1,414 @@
+                problem in enumerate(problems)}
+            perfconfig_to_index = {
+                perfconfig: idx for idx,
+                perfconfig in enumerate(perfconfigs)}
+
+            # Create coverage matrix
+            A = np.zeros((n, m), dtype=int)
+            for problem, perfconfig_list in problems_to_perfconfigs.items():
+                i = problem_to_index[problem]
+                for perfconfig in perfconfig_list:
+                    j = perfconfig_to_index[perfconfig]
+                    A[i][j] = 1
+
+            # Linear programming model to minimize the number of perfconfigs
+            prob = pulp.LpProblem("SetCoverProblems", pulp.LpMinimize)
+            x = pulp.LpVariable.dicts("x", range(m), cat='Binary')
+            prob += pulp.lpSum([x[j]] for j in range(m))
+            for i in range(n):
+                prob += pulp.lpSum([A[i][j] * x[j]
+                                   for j in range(m)]) >= 1, f"Cover_problem_{i}"
+
+            prob.solve(pulp.PULP_CBC_CMD(msg=0))
+
+            selected_configs = [perfconfigs[j]
+                                for j in range(m) if x[j].varValue == 1]
+
+            result[data_type] = selected_configs
+
+        return result
+
+
+def combine_data(input_dir, no_splitK):
+    """
+    Combine all *.debug tuning data into a single file.
+    """
+    tsv_files = glob.glob(os.path.join(input_dir, f"*.debug"))
+
+    dfs = []
+    for file in tsv_files:
+        df = pd.read_csv(file, sep='\t', index_col=None)
+        df = df[df.columns[1:]]
+        dfs.append(df)
+    if not dfs:
+        return None
+    new_df = pd.concat(dfs, ignore_index=True)
+
+    # Remove splitK from tuning data
+    if no_splitK:
+        df_filtered = new_df[new_df['PerfConfig'].str.split(',').str[6] == '1']
+        new_df = df_filtered
+
+    return new_df
+
+
+def print_results(result):
+    """
+    Print selected perfconfigs for each data type.
+    """
+    for datatype, perfconfings in result.items():
+        print(f"Datatype: {datatype}")
+        for idx, perfconfig in enumerate(perfconfings):
+            print(f" {idx + 1}: {perfconfig}")
+
+
+def main(args=None):
+    """
+    usage: quickTunerGen.py [-h] --input-dir INPUT_DIR --op {gemm,conv} [--th TH] --arch ARCH [--update] [--no-splitK]
+    usage exsample: python3 quickTunerGen.py --input-dir tunedData --op conv --arch gfx90a --update --no-splitK
+    """
+    if args is None:
+        args = sys.argv[1:]
+
+    parser = argparse.ArgumentParser(prog='quickTunerGen.py')
+
+    parser.add_argument('--input-dir',
+                        required=True,
+                        type=str)
+
+    parser.add_argument('--op',
+                        required=True,
+                        type=str,
+                        choices=["gemm", "conv"])
+
+    parser.add_argument("--th",
+                        required=False,
+                        type=float,
+                        default=0.93)
```

Comment:
That's correct, but as I understood it, we needed faster quick-tuning, and it was acceptable to sacrifice a few percent of performance. So, I decided to extend on 0.93 since it noticeable reduced the number of configs.